### ISSUE TITLE
File not found errors in gammapy integration test

### ISSUE THREAD

**Issue by HDembinski** [gammapy/gammapy#2144]
Hi, I am trying to use the following script to do an integration test in iminuit for gammapy

```
#!/bin/bash
# run this before a release to check that gammapy is not broken
if [ ! -e gammapy ]; then
  virtualenv gammapy
fi
source gammapy/bin/activate || { echo "Error: You must execute or source this script!"; exit 0; }
pip install cython numpy pytest
pip install .
pip install gammapy pytest-astropy
cd /tmp
gammapy info
python -c "import sys, gammapy; sys.exit(gammapy.test())"
```

but I am getting gammapy-internal errors, such as

```
======================================================================================== ERRORS =========================================================================================
_________________________________________________________________________ ERROR at setup of test_map_maker_ring _________________________________________________________________________

    @pytest.fixture(scope="session")
    def observations():
>       data_store = DataStore.from_dir("$GAMMAPY_DATA/cta-1dc/index/gps/")

/home/hdembinski/Code/iminuit/gammapy/lib/python3.6/site-packages/gammapy/cube/tests/test_make.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'gammapy.data.data_store.DataStore'>, base_dir = PosixPath('$GAMMAPY_DATA/cta-1dc/index/gps')
hdu_table_filename = PosixPath('$GAMMAPY_DATA/cta-1dc/index/gps/hdu-index.fits.gz'), obs_table_filename = PosixPath('$GAMMAPY_DATA/cta-1dc/index/gps/obs-index.fits.gz')

    @classmethod
    def from_dir(cls, base_dir, hdu_table_filename=None, obs_table_filename=None):
        """Create from a directory.
    
        Parameters
        ----------
        base_dir : str, Path
            Base directory of the data files.
        hdu_table_filename : str, Path
            Filename of the HDU index file. May be specified either relative
            to `base_dir` or as an absolute path. If None, the default filename
            will be looked for.
        obs_table_filename : str, Path
            Filename of the observation index file. May be specified either relative
            to `base_dir` or as an absolute path. If None, the default filename
            will be looked for.
        """
        base_dir = make_path(base_dir)
    
        if hdu_table_filename:
            hdu_table_filename = make_path(hdu_table_filename)
            if (base_dir / hdu_table_filename).exists():
                hdu_table_filename = base_dir / hdu_table_filename
        else:
            hdu_table_filename = base_dir / cls.DEFAULT_HDU_TABLE
    
        if obs_table_filename:
            obs_table_filename = make_path(obs_table_filename)
            if (base_dir / obs_table_filename).exists():
                obs_table_filename = base_dir / obs_table_filename
        else:
            obs_table_filename = base_dir / cls.DEFAULT_OBS_TABLE
    
        if not hdu_table_filename.exists():
>           raise IOError("File not found: {}".format(hdu_table_filename))
E           OSError: File not found: $GAMMAPY_DATA/cta-1dc/index/gps/hdu-index.fits.gz

/home/hdembinski/Code/iminuit/gammapy/lib/python3.6/site-packages/gammapy/data/data_store.py:110: OSError
```

How can I fix this?

**Comment by HDembinski** [comment]
@cdeil Any ideas?

**Comment by cdeil** [comment]
I wonder why didn't this come up in CI. Which Gammapy version do you have? Please post output of `gammapy info`.

**Comment by cdeil** [comment]
I would also suggest what Henrii suggested: to have a build or pipeline to run integration tests against packages using iminuit in the cloud, with well-defined scripts and environment. It's a bit of work to set up now, but then avoids work having to create and debug execution environments manually locally.

**Comment by cdeil** [comment]
The test fail you show is for `test_map_maker_ring`. I think there's a `@requires_data` missing here:
https://github.com/gammapy/gammapy/blob/6117dc77b8fe8cef5f689d0de5e00826a0d76cec/gammapy/cube/tests/test_make.py#L113
I can fix that in Gammapy master.

Can you please post the full test log you get in a gist?

**Comment by cdeil** [comment]
I see, there's many cases now of Gammapy tests in master where `@requires_data` is missing.
I'll make a PR next week to add those.

---

Why didn't this issue show up in our CI?

I think because most CI builds have the test data:

https://github.com/gammapy/gammapy/blob/fcf37b53db12eceaf6ed4c40f8102de537156274/.travis.yml#L161-L165

https://github.com/gammapy/gammapy/blob/fcf37b53db12eceaf6ed4c40f8102de537156274/azure-pipelines.yml#L44

There's one build that doesn't fetch the test data:

https://github.com/gammapy/gammapy/blob/fcf37b53db12eceaf6ed4c40f8102de537156274/.travis.yml#L50

I think the simplest way to check for missing `@requires_data` would be to try and change ` SETUP_CMD='egg_info' ` to ` SETUP_CMD='test'` there.

**Comment by cdeil** [comment]
I'm fixing this in #2145 .

Actually this test never ran, because it was listed under `matrix` but then we had another `matrix` in the `.travis.yml`: 
https://github.com/gammapy/gammapy/blob/fcf37b53db12eceaf6ed4c40f8102de537156274/.travis.yml#L50

So instead what I'll do is put `FETCH_GAMMAPY_DATA=false` in this build:
https://github.com/gammapy/gammapy/blob/fcf37b53db12eceaf6ed4c40f8102de537156274/.travis.yml#L74-L77

**Comment by cdeil** [comment]
@HDembinski - This issue should be fixed now in latest Gammapy master (via #2145).
Thanks for reporting it!
